test/e2e-go/features/catchup/basicCatchup_test.go

```diff
@@ -76,19 +77,22 @@ func TestBasicCatchup(t *testing.T) {
 // The current versions are the original v1 and the upgraded to v2.1
 func TestCatchupOverGossip(t *testing.T) {
 	t.Parallel()
+
+	versions := network.SupportedProtocolVersions
+	require.LessOrEqual(t, len(versions), 3)
+
 	// ledger node upgraded version, fetcher node upgraded version
-	runCatchupOverGossip(t, false, false)
-	// ledger node older version, fetcher node upgraded version
-	runCatchupOverGossip(t, true, false)
-	// ledger node upgraded older version, fetcher node older version
-	runCatchupOverGossip(t, false, true)
-	// ledger node older version, fetcher node older version
-	runCatchupOverGossip(t, true, true)
+	for i := 0; i < len(versions); i++ {
```

Comment:
This looks redundant to go over the more recent version (i.e. no downgrade). 
Index 0 is iterator over here to test the code path which loads the version from the configuration file instead of just relying on the default. Moreover, the configuration might have the version set to other than the most recent supported version.